algolia/search/client.go

```diff
@@ -127,3 +127,26 @@ func (c *Client) CustomRequest(
 ) error {
 	return c.transport.Request(&res, method, path, body, k, opts...)
 }
+
+// GetStatus retrieves the task status according to the Algolia engine for the
+// given task.
+func (c *Client) GetStatus(taskID int64, opts ...interface{}) (res TaskStatusRes, err error) {
+	path := c.path("/task/%d", taskID)
+	err = c.transport.Request(&res, http.MethodGet, path, nil, call.Read, opts...)
+	return
+}
+
+// WaitTask blocks until the task identified by the given taskID is completed on
+// Algolia engine.
+func (c *Client) WaitTask(taskID int64, opts ...interface{}) error {
+	return waitWithRetry(func() (bool, error) {
+		res, err := c.GetStatus(taskID, opts...)
+		if err != nil {
+			return true, err
+		}
+		if res.Status == "published" {
```

Comment:
These 4 lines could be `return res.Status == "published", nil`

Reply:
I suppose you meant lines 147-150?
```go
if res.Status == "published" {
  return true, nil
}
return false, nil
```

Reply:
Yes! I'm not sure what happened there :)

Reply:
Done ✅